drivers/bluetooth/hci/hci_da1469x.c

```diff
@@ -0,0 +1,495 @@
+/* hci_da1469x.c - DA1469x CMAC IPC Bluetooth driver */
+
+/*
+ * Copyright (c) 2023 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/drivers/bluetooth/hci_driver.h>
+#include <zephyr/irq.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/random/random.h>
+
+#include <common/bt_str.h>
+
+#include <DA1469xAB.h>
+#include <mbox.h>
+#include <shm.h>
+#include <rand.h>
+
+#define LOG_LEVEL CONFIG_BT_HCI_DRIVER_LOG_LEVEL
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(hci_da1469x);
+
+#define H4_NONE 0x00
+#define H4_CMD  0x01
+#define H4_ACL  0x02
+#define H4_EVT  0x04
+#define H4_ISO  0x05
+
+static K_KERNEL_STACK_DEFINE(rng_thread_stack, CONFIG_BT_RX_STACK_SIZE);
+static struct k_thread rng_thread_data;
+struct k_sem rng_sem;
+
+static K_KERNEL_STACK_DEFINE(rx_thread_stack, CONFIG_BT_RX_STACK_SIZE);
+static struct k_thread rx_thread_data;
+
+static struct {
+	struct net_buf *buf;
+	struct k_fifo   fifo;
+
+	uint16_t    remaining;
+	uint16_t    discard;
+
+	bool     have_hdr;
+	bool     discardable;
+
+	uint8_t     hdr_len;
+
+	uint8_t     type;
+	union {
+		struct bt_hci_evt_hdr evt;
+		struct bt_hci_acl_hdr acl;
+		struct bt_hci_iso_hdr iso;
+		uint8_t hdr[4];
+	};
+} rx = {
+	.fifo = Z_FIFO_INITIALIZER(rx.fifo),
+};
+
+static inline void h4_get_type(void)
```

Comment:
Unless you've profiled the code and concluded that the compiler doesn't do an optimal job, please don't do these explicit inline declarations. Same goes for the other functions here (I won't comment on them separately).